languagetool-server/src/test/java/org/languagetool/server/HTTPServerTest.java

```diff
@@ -313,7 +314,11 @@ private void runDataTests() throws IOException {
   }
 
   @Test
-  public void testTimeout() {
+  public void testTimeout() throws IOException {
+    GermanyGerman lang = new GermanyGerman();
+    // avoid rule <clinit> being interrupted by timeout, resulting in NCDFEs in the following tests
+    lang.getRelevantRules(JLanguageTool.getMessageBundle(), null, null, Collections.emptyList());
```

Comment:
testHTTPServer started to fail because the initialization became lazier and testTimeout's cancellation of futures happened during GermanWordRepeatRule clinit. Such a hack seems OK for the test, but something might be needed to be fixed in production as well. E.g. using more graceful alternatives of `future.cancel`/`shutdownNow` in `TextChecker`.